libcontainer/intelrdt/intelrdt.go

```diff
@@ -166,22 +205,38 @@ type Manager interface {
 
 // This implements interface Manager
 type intelRdtManager struct {
-	mu     sync.Mutex
-	config *configs.Config
-	id     string
-	path   string
+	mu              sync.Mutex
+	config          *configs.Config
+	id              string
+	path            string
+	groupType       string
+	switchGroupType func(m *intelRdtManager, getIntelRdtPath func(string, string) (string, error)) error
```

Comment:
In my opinion, the member function `intelRdtManager.switchGroupType()` is unnecessary. The only caller is `intelRdtManager.Set()` and there are two parameters `m.id` and `m.groupType` are passed into.

We could have a private function `intelrdt.switchGroup()` and call it in `intelRdtManager.Set()`.
```
- func switchGroup(m *intelRdtManager, GetIntelRdtPath func(string, string) (string, error)) error {
+ func switchGroup(id string, grouptype string) error {
````

In `intelRdtManager.Set()`:
```
- if err := m.switchGroupType(m, GetIntelRdtPath); err != nil {
+ if err := switchGroup(m.id, m.groupType); err != nil {
```

Reply:
How about mocking environment in tests?